includes/discovery/entity-physical/entity-physical.inc.php

```diff
@@ -25,6 +26,57 @@
     echo ' hwEntityBomEnDesc';
     $entity_array = snmpwalk_cache_oid($device, 'hwEntityBomEnDesc', $entity_array, 'ENTITY-MIB:HUAWEI-ENTITY-EXTENT-MIB');
 }
+if ($device['os'] == 'saf-cfml4') {
```

Comment:
This is how I would format this section:

```php
if ($device['os'] == 'saf-cfml4') {
    $entity_array = [];
    echo ' saf-cfml4Anatomy';
    $oid = '.1.3.6.1.4.1.7571.100.1.1.2.22';
    $device_array = snmpwalk_cache_oid($device, $oid, [], 'SAF-MPMUX-MIB')[0];

    $entity_array[] = [
        'entPhysicalVendorType' => 'CFM L4',
        'entPhysicalContainedIn' => '0',
        'entPhysicalClass' => 'chassis',
        'entPhysicalParentRelPos' => '-1',
        'entPhysicalName' => 'Chassis',
        'entPhysicalSerialNum' => $device_array['serialNumber'],
        'entPhysicalMfgName' => 'SAF',
        'entPhysicalModelName' => 'CFM L4',
        'entPhysicalIsFRU' => 'true',
    ];

    foreach ([1 => 'rf1Version', 2 => 'rf2Version'] as $index => $item) {
        $entity_array[] = [
            'entPhysicalDescr' => $device_array[$item],
            'entPhysicalVendorType' => 'radio',
            'entPhysicalContainedIn' => 1,
            'entPhysicalClass' => 'module',
            'entPhysicalParentRelPos' => $index,
            'entPhysicalName' => "Radio $index",
            'entPhysicalIsFRU' => 'true',
        ];
    }

    foreach (range(1, 4) as $index) {
        $entity_array[] = [
            'entPhysicalDescr' => 'Module Container',
            'entPhysicalVendorType' => 'containerSlot',
            'entPhysicalContainedIn' => 1,
            'entPhysicalClass' => 'container',
            'entPhysicalParentRelPos' => $index + 2,
            'entPhysicalName' => "Slot $index",
            'entPhysicalIsFRU' => 'false',
        ];
    }

    foreach ([1 => 'm1Description', 2 => 'm2Description', 3 => 'm3Description', 4 => 'm4Description'] as $index => $item) {
        if (!str_contains('N/A', $device_array[$item])) {
            $entity_array[] = [
                'entPhysicalDescr' => $device_array[$item],
                'entPhysicalVendorType' => 'module',
                'entPhysicalContainedIn' => $index + 3,
                'entPhysicalClass' => 'module',
                'entPhysicalParentRelPos' => 1,
                'entPhysicalName' => "Module $index",
                'entPhysicalIsFRU' => 'true',
            ];
        }
    }
}
```

Reply:
Thanks for this suggestion, I did not know how to do "snmpwalk_cache_oid($device, $oid, [], 'SAF-MPMUX-MIB')**[0]**". I used your code and corrected it a bit and added an extra condition as not all cfm devices have modular cards.